desktop-widgets/locationinformation.cpp

```diff
@@ -313,15 +313,18 @@ void LocationInformationWidget::on_diveSiteDistance_textChanged(const QString &s
 
 void LocationInformationWidget::reverseGeocode()
 {
-	dive_site *ds = diveSite; /* Use local copy in case the object-copy is NULL'd by another thread while we perform the geo lookup */
+	bool buttonWasEnabled = ui.geoCodeButton->isEnabled();
+	ui.geoCodeButton->setEnabled(false);
+	dive_site *ds = diveSite; /* Save local copy; possibility of user closing the widget while reverseGeoLookup is running (see #2930) */
 	location_t location = parseGpsText(ui.diveSiteCoordinates->text());
-	if (!ds || !has_location(&location))
-		return;
 	taxonomy_data taxonomy = { 0, 0 };
-	reverseGeoLookup(location.lat, location.lon, &taxonomy);
-	if (ds != diveSite)
-		return;
-	Command::editDiveSiteTaxonomy(ds, taxonomy);
+	if (ds && has_location(&location)) {
+		reverseGeoLookup(location.lat, location.lon, &taxonomy);
+	}
+	if(ds == diveSite) {
```

Comment:
This looks like a new possible null-pointer access.

Suppose the function is entered with `diveSite == nullptr`.
- `ds` is set to null.
- no geo lookup is performed.
- but the condition 'ds == diveSite` is true and it will call the edit command with a nullptr, isn't it?

Reply:
Good catch, bug introduced due to refactoring for disabling the button.

My original refactor replaced the "return" with "goto done" but figured that probably wasn't an acceptable coding style in this project.